### PR TITLE
[13.0][MIG] hr_payroll: migrate from odoo v12

### PR DESCRIPTION
**Pull request by davejames**
The hr_payroll module was dropped from odoo from v13 onwards. This PR migrates it to v13.

There is another migration attempt in this repo but it has some issues and hasn't seen a commit in over 3 months. This PR also benefits from fixes which have been made in the odoo v12 repo over the last year which are not included in the other PR.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
There are 2 previous attempts, but your is the first one green.

@realsaiko @Saran440 do you agree on switching to this one?

@davejames some tweaks are needed:

- You need to add an initial commit with committer "info@odoo.com" (not you), with the verbatim copy of the v12 module, then apply the subsequent commits with pre-commit + migration.
- You need to add in the README the disclaimer mentioned here: https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#backporting-odoo-modules

**Comment by davejames** [comment]
@pedrobaeza I have made the changes but this is my first time doing an OCA style readme, it is my understanding that the template will be replaced by the contents of the readme directory. Can you please confirm I've done that correctly?

**Comment by pedrobaeza** [comment]
Yes, correct. You can pre-generate it if you want, but it's OK this way.

**Comment by Saran440** [comment]
Yes, I agree switching to this one.

File: hr_payroll/data/hr_payroll_demo.xml [hunk 1]
```diff
@@ -0,0 +1,148 @@
+<?xml version="1.0" encoding="utf-8" ?>
```
Comment (FernandoRomera) [inline hr_payroll/data/hr_payroll_demo.xml]: Can you post this file in demo directory?
This are the guidelines: https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#directories

Comment (davejames) [reply hr_payroll/data/hr_payroll_demo.xml]: done

**Comment by davejames** [comment]
I have made some changes after reviewing the contribution guidelines. I have not squashed the commits for ease of keeping track of the changes during reviewing, but will squash them upon approval.

**Comment by davejames** [comment]
OK I think this is ready for further review now.

I have split out all the models into separate files, same for the views. Then I have also renamed the views as per the OCA guidelines.

This was always a difficult module to read due to the long files with many models in them, hopefully it's a bit clearer now.

File: hr_payroll/security/hr_payroll_security.xml [hunk 2]
```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0" encoding="utf-8" ?>
```
Comment (flachica) [inline hr_payroll/security/hr_payroll_security.xml:1]: The model hr.contribution.register has the company_id but I can't see the rule like described [here](https://www.odoo.com/documentation/13.0/howtos/company.html#security-rules). There are more models with these field without the rule

**Review by flachica — CHANGES_REQUESTED** [review]

**Review by flachica — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@realsaiko @Saran440 can you please review this for merging a initial good base?

File: hr_payroll/security/hr_payroll_security.xml (hunk 2 above)
Comment (davejames) [reply hr_payroll/security/hr_payroll_security.xml:1]: thanks, that's a good point, I added those rules now

**Comment by Saran440** [comment]
Functional Test. Create new database and demo data

@davejames I found bug something when create payslip from employee or batch
Selection_001

and I not found function get_work_days_data in core odoo should it _get_work_days_data or I miss something?

**Comment by davejames** [comment]
> and I not found function get_work_days_data in core odoo should it _get_work_days_data or I miss something?

well spotted, this method is renamed in 13.0. I have corrected it

**Comment by Saran440** [comment]
@davejames this point is fixed. and Functional test create payroll from payroll batch look good.
but other ways not compute worked days

Step to tested :
- Go to Employee > selected 1 employee and click smart button payslips > create **(NOT computed worked days)**
- Go to Payroll > Employee Payslips > create > selecte employee **(NOT computed worked days)**
- Go to Payslip Batches > create > select employee **(computed worked days)**

Could you check please?

**Comment by davejames** [comment]
Thanks @Saran440 that's some great testing. I can reproduce this on odoo v12 runbot also. Fixed in my latest commit.

**Review by Saran440 — APPROVED** [review]
Functional Test. LGTM :+1:

**Comment by Saran440** [comment]
@pedrobaeza Should this PR squash?

**Comment by pedrobaeza** [comment]
Oh, yes! Aborting the merge.

**Comment by pedrobaeza** [comment]
@davejames please squash your migration commits.

**Comment by davejames** [comment]
@pedrobaeza squash is done

**Comment by pedrobaeza** [comment]
Thanks

/ocabot merge patch

**Pull request merged**